clouddriver-core/src/main/groovy/com/netflix/spinnaker/clouddriver/orchestration/ApplicationContextAtomicOperationsRegistry.groovy

```diff
@@ -31,12 +32,22 @@ class ApplicationContextAtomicOperationsRegistry implements AtomicOperationsRegi
   ApplicationContext applicationContext
 
   @Override
-  AtomicOperationConverter getAtomicOperationConverter(String description, String cloudProvider) {
-    (AtomicOperationConverter) applicationContext.getBean(description)
+  AtomicOperationConverter getAtomicOperationConverter(String description, String cloudProvider, ProviderVersion version) {
+    def result = (AtomicOperationConverter) applicationContext.getBean(description)
+    if (!result.acceptsVersion(version)) {
```

Comment:
Since the helper function would directly throw the exception, could even do this:

`return VersionHelper.verifyMatch(applicationContext.getBean(description), version, description)`
`return VersionHelper.verifyMatch(applicationContext.getBean(validator), version, validator)`

Thought it would feel weird to pass the `applicationContext`, so just inlined that call above.